compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -957,6 +971,15 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       case ctor: MethodSymbol if ctor.isPrimaryConstructor => ctor
     }
 
+  /** Identify `ddef` accessor that have the shape described in issue #1899.
+   *  Note: This is an over approximation, read comment at call site.
+   */
+  private def isAnonymousJSObjectAccessorMethod(ddef: DefDef): Boolean = {
+    val sym = ddef.symbol
+    val cls = sym.owner
+    sym.isPrivate && !sym.isPrivateLocal &&
+    cls.isAnonymousClass && cls.parentSymbols.contains(JSObjectClass)
```

Comment:
The last condition is wrong. The same treatment must also be applied to subclasses of other classes extending `js.Any`.

The appropriate test would be taken care of by `enclosingOwner is RawJSType`